### PR TITLE
Check if Authenticator is loading before attempting to login

### LINKED ISSUE #31: Auto login from localstorage session does not work with Scatter
When I login with the MockAuthenticator and then refresh the page it automatically logs me back in. When I try to do the same with scatter, it gives the following error after refreshing the page and I have to manually login again with scatter:

```
Scatter.js:103 Uncaught (in promise) UALScatterError: Unable to login
    at Scatter.eval (webpack:///./node_modules/ual-scatter/dist/Scatter.js?:103:23)
    at Generator.next (<anonymous>)
    at eval (webpack:///./node_modules/ual-scatter/dist/Scatter.js?:7:71)
    at new Promise (<anonymous>)
    at __awaiter (webpack:///./node_modules/ual-scatter/dist/Scatter.js?:3:12)
    at Scatter.login (webpack:///./node_modules/ual-scatter/dist/Scatter.js?:93:16)
    at UALJs.eval (webpack:///../dist/UALJs.js?:98:45)
    at Generator.next (<anonymous>)
    at eval (webpack:///../dist/UALJs.js?:7:71)
    at new Promise (<anonymous>)
```
What is the best way to remain logged in after refreshing the page with scatter?

### PR DESCRIPTION
**Pull request by TaraTritt**
## Change Description

Session persistence was broken due to attempts to login before the Authenticator was ready. By checking that the Authenticator is not loading before attempting to login, session persistence is now functional.  Fixes #31

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: src/UALJs.test.ts [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
-import { User } from 'universal-authenticator-library'
+import promisePolyFill from 'promise'
+import { Authenticator, User } from 'universal-authenticator-library'
 import { UALJs } from './UALJs'
 
 import { AutologinAuthenticator } from './AuthMocks/AutologinAuthenticator'
 import { BaseMockAuthenticator } from './AuthMocks/BaseMockAuthenticator'
 
+jest.useFakeTimers()
+global.Promise = promisePolyFill
```
Comment (wpwood) [inline src/UALJs.test.ts]: As loathe as I am to suggests comments, I think one would be appropriate here, perhaps with a link to the Jest Github issue explaining why you had to use this.

File: src/UALJs.test.ts [hunk 2]
```diff
@@ -185,37 +184,53 @@ describe('Authenticators', () => {
...
+  // jest.useFakeTimers() changes the order in which promises are run
+  // https://github.com/facebook/jest/pull/6876
+  // the code below along with the global.Promise pollyfill
+  // are workarounds to ensure promises are run in their intended order
+  Promise.resolve().then(() => jest.advanceTimersByTime(1))
+  await sleep(1)
+}
+
+const createNewUALJs = (authenticator: Authenticator, containerElement: HTMLElement) => {
+  return new UALJs(
```
Comment (wpwood) [inline src/UALJs.test.ts]: Use the `=> (` shorthand here

File: src/UALJs.ts [hunk 3]
```diff
@@ -131,6 +131,8 @@ export class UALJs extends UAL {
     localStorage.setItem(UALJs.SESSION_EXPIRATION_KEY, `${thirtyDaysFromNow.getTime()}`)
     localStorage.setItem(UALJs.SESSION_AUTHENTICATOR_KEY, authenticator.constructor.name)
 
+    await this.isAuthenticatorLoading(authenticator)
```
Comment (wpwood) [inline src/UALJs.ts]: If this function is just to wait for loading, perhaps `await this.waitForLoading(authenticating)` or something like that?

File: src/UALJs.test.ts (hunk 1 above)
Comment (TaraTritt) [reply src/UALJs.test.ts]: I had a link in the helper function `runPromises` at the end of the tests, but i agree i can add it here as well for clarity

File: src/UALJs.test.ts (hunk 2 above)
Comment (TaraTritt) [reply src/UALJs.test.ts]: 👍

File: src/UALJs.ts (hunk 3 above)
Comment (TaraTritt) [reply src/UALJs.ts]: 👍updated to `waitForAuthenticatorToLoad`

**Pull request merged**